### DISCUSSION TITLE
Does b2nd_append return only after appending the data?

### DISCUSSION THREAD

**Discussion by leoyala** [Blosc/c-blosc2#600] [Q&A] [answered]
I was wondering whether `b2nd_append` only returns after the data is indeed appended to the array, or if it somehow only places the data in a thread and then returns immediately before the data is actually compressed and written to the file. the same questions could be mentioned for other methods like `b2nd_insert`.

**Comment by FrancescAlted** [comment, marked as answer]
Yes, both return when data has been consolidated in the b2nd array.
